src/databricks/labs/ucx/aws/credentials.py

```diff
@@ -0,0 +1,213 @@
+import logging
+from dataclasses import dataclass
+
+from databricks.labs.blueprint.installation import Installation
+from databricks.labs.blueprint.tui import Prompts
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors.platform import InvalidParameterValue
+from databricks.sdk.service.catalog import (
+    AwsIamRole,
+    Privilege,
+    StorageCredentialInfo,
+    ValidationResultResult,
+)
+
+from databricks.labs.ucx.assessment.aws import (
+    AWSResourcePermissions,
+    AWSResources,
+    AWSRoleAction,
+)
+from databricks.labs.ucx.config import WorkspaceConfig
+from databricks.labs.ucx.framework.crawlers import StatementExecutionBackend
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class AWSStorageCredentialValidationResult:
+    name: str
+    role_arn: str
+    validated_on: str
+    read_only: bool
+    failures: list[str] | None = None
+
+
+class AWSStorageCredentialManager:
```

Comment:
```suggestion
class CredentialManager:
```
keep it DRY - we're already in `aws` package